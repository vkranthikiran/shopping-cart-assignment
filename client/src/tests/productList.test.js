import { fireEvent, render, screen, waitFor } from "@testing-library/react"
import Products from "../components/ProductsList"
import '@testing-library/jest-dom'
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const items = [
    {
        "name": "Fresho Kiwi - Green, 3 pcs",
        "imageURL": "/static/images/products/fruit-n-veg/kiwi-green.jpg",
        "description": "Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.",
        "price": 87,
        "stock": 50,
        "category": "5b6899953d1a866534f516e2",
        "sku": "fnw-kiwi-3",
        "id": "5b6c6a7f01a7c38429530883"
    },
];
let store;
store = mockStore({ products: { products: [...items] }});
function renderWithProviders(ui, { reduxState } = {}) {
    return render(<Router><Provider store={store}>{ui}</Provider></Router>);
}
test('it should', async () => {
    const {getAllByTestId } = await renderWithProviders(<Products />, {
        store: { products: { products: [...items] } }
    });
    getAllByTestId('productCard');
    getAllByTestId('productImage');
    getAllByTestId('productName');
    getAllByTestId('productDesc');
    getAllByTestId('productPrice')
})