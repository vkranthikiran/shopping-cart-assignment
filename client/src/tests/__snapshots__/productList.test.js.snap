// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="product_list"
        id="product_list"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-md-3 p-0 pl-2 mb-3"
          >
            <div
              class="card"
              data-testid="5b6c6a7f01a7c38429530883"
            >
              <div
                class="card-header bg-white"
              >
                <p
                  class="title"
                  data-testid="Fresho Kiwi - Green, 3 pcs"
                >
                   
                  <strong>
                    Fresho Kiwi - Green, 3 pcs
                  </strong>
                </p>
              </div>
              <img
                alt="Fresho Kiwi - Green, 3 pcs"
                class="card-img-top p-2"
                data-testid="productImage"
                src="/static/images/products/fruit-n-veg/kiwi-green.jpg"
              />
              <div
                class="card-body"
              >
                <p
                  class="card-text"
                  data-testid="productDesc"
                >
                  Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.
                </p>
              </div>
              <div
                class="card-footer bg-white"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-sm-6"
                  >
                    <small
                      data-testid="productPrice"
                    >
                      <strong>
                        MRP Rs.
                        87
                      </strong>
                    </small>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        data-testid="product_list"
        id="product_list"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-md-3 p-0 pl-2 mb-3"
          >
            <div
              class="card"
              data-testid="5b6c6a7f01a7c38429530883"
            >
              <div
                class="card-header bg-white"
              >
                <p
                  class="title"
                  data-testid="Fresho Kiwi - Green, 3 pcs"
                >
                   
                  <strong>
                    Fresho Kiwi - Green, 3 pcs
                  </strong>
                </p>
              </div>
              <img
                alt="Fresho Kiwi - Green, 3 pcs"
                class="card-img-top p-2"
                data-testid="productImage"
                src="/static/images/products/fruit-n-veg/kiwi-green.jpg"
              />
              <div
                class="card-body"
              >
                <p
                  class="card-text"
                  data-testid="productDesc"
                >
                  Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.
                </p>
              </div>
              <div
                class="card-footer bg-white"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-sm-6"
                  >
                    <small
                      data-testid="productPrice"
                    >
                      <strong>
                        MRP Rs.
                        87
                      </strong>
                    </small>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="product_list"
      id="product_list"
    >
      <div
        class="row align-items-center"
      >
        <div
          class="col-md-3 p-0 pl-2 mb-3"
        >
          <div
            class="card"
            data-testid="5b6c6a7f01a7c38429530883"
          >
            <div
              class="card-header bg-white"
            >
              <p
                class="title"
                data-testid="Fresho Kiwi - Green, 3 pcs"
              >
                 
                <strong>
                  Fresho Kiwi - Green, 3 pcs
                </strong>
              </p>
            </div>
            <img
              alt="Fresho Kiwi - Green, 3 pcs"
              class="card-img-top p-2"
              data-testid="productImage"
              src="/static/images/products/fruit-n-veg/kiwi-green.jpg"
            />
            <div
              class="card-body"
            >
              <p
                class="card-text"
                data-testid="productDesc"
              >
                Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.
              </p>
            </div>
            <div
              class="card-footer bg-white"
            >
              <div
                class="row align-items-center"
              >
                <div
                  class="col-sm-6"
                >
                  <small
                    data-testid="productPrice"
                  >
                    <strong>
                      MRP Rs.
                      87
                    </strong>
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
